### ISSUE TITLE
Style/Next autocorrection breaks indentation in some cases

### ISSUE THREAD

**Issue by deivid-rodriguez** [bbatsov/rubocop#2450]
The following file

```ruby
def a
  res = nil

  [1, 2, 3].each do |number|
    if number > 2
      if number == 1
        res = "it's one"
      else
        res = "it's not one"
      end
    end
  end

  res
end
```

gets autocorrected to

```ruby
def a
  res = nil

  [1, 2, 3].each do |number|
    next unless number > 2
    if number == 1
      res = "it's one"
    else
      res = "it's not one"
          end
  end

  res
end
```

introducing a `Lint/EndAlignment` offense.

**Comment by lumeet** [comment]
I think this has been fixed on master with #2429. #2453 wouldn't break it either.

**Comment by deivid-rodriguez** [comment]
With both #2429 and #2453 I still get an incorrect autocorrection

```ruby
With def a
  res = nil

  [1, 2, 3].each do |number|
    next unless number > 2
    res = if number == 1
            "it's one"
          else
            "it's not one"
    end
  end

  res
end

**Comment by lumeet** [comment]
Unless I'm completely wrong, `Style/Next` is not to blamed for this. Using the default configuration with `Next` disabled, I'm getting similar alignment for the `end`.

**Comment by jonas054** [comment]
See [this comment](https://github.com/bbatsov/rubocop/issues/1348#issuecomment-56837476) for an explanation of why the `end`s don't get corrected. In this case, not correcting `end` alignment by default is not a good feature, since other cops have misaligned them during auto-correction.

**Comment by deivid-rodriguez** [comment]
@jonas054 So what would be the best solution here? Maybe allowing autocorrections which leave offenses in the corrected code (with the expectation that they will be corrected by other cops) is not a good idea?

**Comment by jonas054** [comment]
That's what we usually do, but `end`s are special -- they might not be auto-corrected, depending on configuration. So if they were correctly aligned to begin with, it would be much better if `Style/Next` could make sure that they continued to be correctly aligned.

**Comment by deivid-rodriguez** [comment]
So then we should consider [this](https://github.com/lumeet/rubocop/commit/1855ccec712e42206624359a310509e4c6f1debe#commitcomment-14598788) a bug. Right?

**Comment by jonas054** [comment]
Yes, I think so.

**Comment by alexdowad** [comment]
I'm about to push a fix for this one to my open PR.
